### PR TITLE
Add plugin server liveness probe

### PR DESCRIPTION
**Pull request by yakkomajuri**
Again, chart n00b over here.

But PRs over issues. I think this is what's needed to leverage the new plugin server health endpoint once https://github.com/PostHog/posthog/pull/7983 is merged

### CODE REVIEW

File: charts/posthog/templates/plugins-deployment.yaml [hunk 1]
```diff
@@ -148,6 +148,26 @@ spec:
 {{- if .Values.plugins.env }}
 {{ toYaml .Values.plugins.env | indent 8 }}
 {{- end }}
+        livenessProbe:
+          failureThreshold: {{ .Values.plugins.livenessProbe.failureThreshold }}
+          httpGet:
+            path: /health
```
Comment (yakkomajuri) [inline charts/posthog/templates/plugins-deployment.yaml:154]:
```suggestion
            path: /_health
```

**Review by fuziontech — APPROVED** [review]
nice fix - looks good!
🚢

File: charts/posthog/templates/plugins-deployment.yaml [hunk 2]
```diff
@@ -148,6 +148,26 @@ spec:
 {{- if .Values.plugins.env }}
 {{ toYaml .Values.plugins.env | indent 8 }}
 {{- end }}
+        livenessProbe:
+          failureThreshold: {{ .Values.plugins.livenessProbe.failureThreshold }}
+          httpGet:
+            path: /_health
+            port: 5000
```
Comment (yakkomajuri) [inline charts/posthog/templates/plugins-deployment.yaml:155]:
```suggestion
            port: 6738
```

**Comment by guidoiaquinti** [comment]
Sorry for the late review here @yakkomajuri . A couple of things:

1. it is an anti-pattern to use the same endpoint for liveness and readiness probes (see this [nice PR](https://github.com/PostHog/posthog/pull/8258) from @hazzadous as example) and the official k8s [doc](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/).
1. is the `/_health` endpoint in the `plugin-server` already available in `1.32`? If not, we should wait till 1.33 is released and referenced as default in this chart.

File: charts/posthog/values.yaml [hunk 3]
```diff
@@ -180,7 +180,33 @@ plugins:
...
+    # -- The liveness probe failure threshold
+    failureThreshold: 5
+    # -- The liveness probe initial delay seconds
+    initialDelaySeconds: 50
+    # -- The liveness probe period seconds
+    periodSeconds: 10
+    # -- The liveness probe success threshold
+    successThreshold: 1
+    # -- The liveness probe timeout seconds
+    timeoutSeconds: 5
```
Comment (yakkomajuri) [inline charts/posthog/values.yaml:198]:
```suggestion
```

File: charts/posthog/templates/plugins-deployment.yaml [hunk 4]
```diff
@@ -104,6 +104,26 @@ spec:
...
+        livenessProbe:
+          failureThreshold: {{ .Values.plugins.livenessProbe.failureThreshold }}
+          httpGet:
+            path: /_health
+            port: 6738
+            scheme: HTTP
+          initialDelaySeconds: {{ .Values.plugins.livenessProbe.initialDelaySeconds }}
+          periodSeconds: {{ .Values.plugins.livenessProbe.periodSeconds }}
+          successThreshold: {{ .Values.plugins.livenessProbe.successThreshold }}
+          timeoutSeconds: {{ .Values.plugins.livenessProbe.timeoutSeconds }}
```
Comment (yakkomajuri) [inline charts/posthog/templates/plugins-deployment.yaml:116]:
```suggestion
```

File: charts/posthog/values.yaml [hunk 5]
```diff
@@ -180,7 +180,22 @@ plugins:
...
   affinity: {}
-
+  # priorityClassName: ""
+  # schedulerName:
+  # Optional extra labels for pod, i.e. redis-client: "true"
+  # podLabels: []
+  # concurrency:
+  readinessProbe:
+    # -- The readiness probe failure threshold
+    failureThreshold: 10
```
Comment (yakkomajuri) [inline charts/posthog/values.yaml:190]:
```suggestion
    failureThreshold: 15
```

**Comment by guidoiaquinti** [comment]
> Removed health for now, but readiness should be available from 1.34.0 onwards, would love to have it in then @guidoiaquinti

👍 in that case we need to wait till the default image gets to `1.34`

**Comment by hazzadous** [comment]
I going to close this to reduce the noise in this repo, if it's really important feel free to open again.

**Pull request closed without merging**